### PR TITLE
chore: tests rewritten with gaiad blockchain as a host

### PR DESCRIPTION
**Pull request by swelf19**
implements: https://p2pvalidator.atlassian.net/browse/NTRN-118

linked prs:
https://github.com/neutron-org/neutron-query-relayer/pull/39 added env variable
https://github.com/neutron-org/neutron-contracts/pull/34 added field with denom in the delegate msg

1) moved root entry point to run the test environment from the `neutron` repo to the `setup` dir of the tests repo
2) every part of the environment runs in a separate docker.
3) we use gaiad (cosmos-hub) as a host network in the tests

to build all necessary parts for tests please see the readme

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -28,10 +38,11 @@ yarn test:interchain_kv_query # interchain tx query test
 ## Environment variables you can redefine
 
 ```
-NEUTRON_DIR - directory where Neutron is located
-DENOM - denom used for tests
+NEUTRON_DENOM - neutron denom
+COSMOS_DENOM - host network denom
```
Comment (oopcode) [inline README.md:42]: host? either provider or cosmos/gaia, i guess

File: setup/docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,81 @@
+version: '3.8'
+services:
+  neutron-node:
+    image: neutron-node
+    container_name: neutron-node
+    volumes:
+      - data:/opt/neutron/data
+      - /home/swelf/src/lido/neutron-contracts/:/opt/neutron/contracts
```
Comment (oopcode) [inline setup/docker-compose.yml:8]: /home/swelf

File: README.md [hunk 3]
```diff
@@ -28,10 +38,11 @@ yarn test:interchain_kv_query # interchain tx query test
...
 CONTRACTS_PATH - path to contracts that will be used in tests
-ADDRESS_PREFIX - address prefix
+NEUTRON_ADDRESS_PREFIX - address prefix for neutron controller network
+COSMOS_ADDRESS_PREFIX - address prefix for gaia(cosmoshub) host network
```
Comment (oopcode) [inline README.md:45]: `gaia (cosmoshub)`

**Review by oopcode — APPROVED** [review]

**Review by pr0n00gler — APPROVED** [review]
Tested, everything works fine

**Pull request merged**